job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -514,10 +515,47 @@ class WebApi(system: ActorSystem,
    *    GET /healthz              - return OK or error message
    */
   def healthzRoutes: Route = pathPrefix("healthz") {
+    import ContextSupervisor._
+    import JobManagerActor._
+    import JobInfoActor._
+    import JobDAOActor._
     //no authentication required
     get { ctx =>
       logger.info("Receiving healthz check request")
-      ctx.complete("OK")
+      var actorsAlive: Int = 0
+      val supervisorFuture = (supervisor ? GetContext("dummycontext")).mapTo[NoSuchContext.type]
+      val jobInfofuture = (jobInfoActor ? GetJobStatus("dummyjobid")).mapTo[NoSuchJobId.type]
+      val jobDaofuture = (daoActor ? GetJobInfos(1)).mapTo[JobInfos]
+      val statusActor = system.actorOf(JobStatusActor.props(daoActor))
+      val jobStatusfuture = (statusActor ? Unsubscribe("dummyjobid", null)).mapTo[NoSuchJobId.type]
+      val resultActor = GetResultActor("dummycontext")
+      val jobResultfuture = (jobInfoActor ? GetJobResult("dummyjobid")).mapTo[NoSuchJobId.type]
+
+      val listOfFutures = Seq(supervisorFuture, jobInfofuture, jobDaofuture, jobStatusfuture, jobResultfuture)
+      val futureOfList = Future.sequence(listOfFutures)
+
+      try {
+        val results = Await.result(futureOfList, 60 seconds)
```

Comment:
Our general spray timeout is 40 seconds (in application.conf `request-timeout = 40 s`). 
Also I think it's enough to have 10-15 seconds timeout here. It's a general timeout in Jobserver and futures are executed in parallel, so 6 futures shouldn't make timeout 6 times longer.

Reply:
I reduced the timeout to 30 seconds.